Objects/abstract.c

```diff
@@ -2881,7 +2881,47 @@ PyAIter_Check(PyObject *obj)
             tp->tp_as_async->am_anext != &_PyObject_NextNotImplemented);
 }
 
+static int
+iternext(PyObject *iter, PyObject **item)
+{
+    *item = (*Py_TYPE(iter)->tp_iternext)(iter);
```

Comment:
I suggest to get the attribute on a separated line (I didn't test this code):

```suggestion
    iternextfunc tp_iternext = Py_TYPE(iter)->tp_iternext;
    *item = tp_iternext(iter);
```

Reply:
I considered this, but I abandoned it in hope that it would create a smaller diff (it didn't). I'll reconsider.